src/wd.rs

```diff
@@ -0,0 +1,152 @@
+//! WebDriver types and declarations.
+
+use crate::error;
+use std::borrow::Cow;
+use std::convert::TryFrom;
+use std::fmt;
+
+/// A [handle][1] to a browser window.
+///
+/// Should be obtained it via [`Client::window()`] method (or similar).
+///
+/// [1]: https://w3.org/TR/webdriver/#dfn-window-handles
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct WindowHandle(String);
+
+impl From<WindowHandle> for String {
+    fn from(w: WindowHandle) -> Self {
+        w.0
+    }
+}
+
+impl<'a> TryFrom<Cow<'a, str>> for WindowHandle {
+    type Error = error::InvalidWindowHandle;
+
+    /// Makes the given string a [`WindowHandle`].
+    ///
+    /// Avoids allocation if possible.
+    ///
+    /// # Errors
+    ///
+    /// If the given string is [`"current"`][1].
+    ///
+    /// [1]: https://w3.org/TR/webdriver/#dfn-window-handles
+    fn try_from(s: Cow<'a, str>) -> Result<Self, Self::Error> {
+        if s != "current" {
```

Comment:
I'm not sure I follow why this is disallowed for _all_ instances of `WindowHandle`? I thought it was permitted as an argument for selecting a window?

Reply:
Just following [the spec stated in docs](https://www.w3.org/TR/webdriver/#dfn-window-handles) here:

> Each browsing context has an associated window handle which uniquely identifies it. This must be a String and must not be "current".

Reply:
Ah, interesting. I thought a window handle could also be passed as an argument to certain commands, in which case "current" would be handled as a special case (that feels like why they'd have a carve-out for that string), but can't actually find any commands that do that at the moment, so all good then :+1: